### DISCUSSION TITLE
madis2nc no longer compatible with mesonet data

### DISCUSSION THREAD

**Discussion by scatt11** [dtcenter/METplus#1189] [File I/O] [answered]
The madis2nc tool now rejects mesonet data from MADIS due to a change in variables/format to the NetCDF files.  At some point in 2016, the mesonet output files were updated and no longer contain variables such as precip3hr, precip6hr, etc.  An error is now thrown by the madis2nc tool (see below).  I have contacted the MADIS support team and confirmed this was an intentional update.  That is that precip3hr and other missing variables should not be variables in the new mesonet netCDF (files had 239 variables and were reduced 219).  Further, they mentioned that they do add/change variables to the raw data from time to time, and that may complicate the current methods for identifying MADIS data type.  I'm happy to forward this email chain along to someone on the MET team for documentation.  Lastly, the change in data files can be demonstrated in the following files:
https://madis-data.ncep.noaa.gov/madisPublic1/data/archive/2016/01/01/LDAD/mesonet/netCDF/20160101_0000.gz
https://madis-data.ncep.noaa.gov/madisPublic1/data/archive/2017/01/01/LDAD/mesonet/netCDF/20170101_0000.gz
![image](https://user-images.githubusercontent.com/85969291/135931637-a8780bf6-fed3-4909-84f7-d0ebd631aca9.png)

If a workaround is possible so that MET can ingest mesonet data from MADIS, that would be fantastic.  Otherwise I would suggest an update for a future release.

Thank you very much MET team!

**Comment by jprestop** [comment, marked as answer]
Hi @scatt11.  Thank you for bringing this to our attention. I was able to confirm this behavior using the met-10.1.0-beta2 release.  I created an [issue #1936](https://github.com/dtcenter/MET/issues/1936) to fix the problem in the main_v10.0 branch for MET version 10.0.1 and in the develop branch for MET version 10.1.0.  

I assigned the issue to the met-10.1.0-beta4 release which is scheduled to come out in mid-November.  You are welcome to pursue Python embedding as a means for a workaround in the meantime, if you’d like. Here is a link to [Appendix F Python Embedding](https://met.readthedocs.io/en/latest/Users_Guide/appendixF.html) in our User’s Guide to help get you started.

**Reply by jprestop** [reply]
Hi @scatt11.  I just wanted to follow up and let you know that I'd like to mark the above response as the "Answer" and lock this conversation to prevent future posts. Our team has decided that once discussions have been answered, and the answer has been confirmed, we'll lock them. We want to encourage users to ask new questions in new discussions rather than posting to old ones. Hopefully that'll make the questions/answers easier for other users to follow. So if/when more issues/questions arise, please feel free to start a new discussion.

Could you please confirm that you feel this question has been answered appropriately?

**Reply by scatt11** [reply]
Yes thank you for the prompt response, and for adding this fix to the list.

**Reply by jprestop** [reply]
Great! Thank you very much for the quick confirmation!
